tests/python_package_test/test_engine.py

```diff
@@ -938,6 +938,52 @@ def test_early_stopping_via_global_params(first_metric_only):
     assert "error" in gbm.best_score[valid_set_name]
 
 
+@pytest.mark.parametrize("early_stopping_round", [-10, -1, 0, None, "None"])
+def test_early_stopping_non_positive_values(early_stopping_round):
```

Comment:
```suggestion
def test_early_stopping_is_not_enabled_for_non_positive_stopping_rounds(early_stopping_round):
```

Would you please consider this suggestion to make the intent of the test a bit clearer?

Reply:
Thanks for this, I was struggling a bit to balance clarity with length-of-function-name, but I guess longer is better.